### PR TITLE
Improve video playback performance

### PR DESCRIPTION
**Pull request by karjonas**

### CODE REVIEW

File: tide/core/data/FFMPEGMovie.cpp [hunk 1]
```diff
@@ -177,59 +166,52 @@ double FFMPEGMovie::getFrameDuration() const
...
-
 PicturePtr FFMPEGMovie::getFrame(double posInSeconds)
 {
     posInSeconds = std::max(0.0, std::min(posInSeconds, getDuration()));
+    const auto frameDuration = _videoStream->getFrameDuration();
+    const auto target = std::max(0.0, posInSeconds - frameDuration);
+    const auto frameIndex = _videoStream->getFrameIndex(target);
+
+    _frameIndex = frameIndex;
+    _streamPosition = posInSeconds;
```
Comment (tribal-tec) [inline tide/core/data/FFMPEGMovie.cpp]: not needed

File: tide/core/data/FFMPEGMovie.cpp [hunk 2]
```diff
@@ -177,59 +166,52 @@ double FFMPEGMovie::getFrameDuration() const
...
-        const auto frameDuration = _videoStream->getFrameDuration();
-        const auto target = std::max(0.0, posInSeconds - frameDuration);
-        const auto frameIndex = _videoStream->getFrameIndex(target);
         if (!_videoStream->seekToNearestFullframe(frameIndex))
-            return PicturePtr();
+            return nullptr;
     }
 
-    const auto targetTimestamp = _videoStream->getTimestamp(posInSeconds);
+    const auto targetTimestamp = _videoStream->getTimestamp(frameIndex);
```
Comment (tribal-tec) [inline tide/core/data/FFMPEGMovie.cpp]: still posInSeconds?

File: tide/core/data/FFMPEGMovie.cpp (hunk 1 above)
Comment (karjonas) [reply tide/core/data/FFMPEGMovie.cpp]: It is actually used by FFMPEGMovie::getPosition()

File: tide/core/data/FFMPEGMovie.cpp (hunk 2 above)
Comment (karjonas) [reply tide/core/data/FFMPEGMovie.cpp]: There are two getTimestamp methods. One that takes a double(seconds) and one that takes an int(frame). I removed the double one.

**Comment by ppodhajski** [comment]
retest this please

**Pull request merged**